packages/vm/src/modules/file_system_cache.rs

```diff
@@ -65,6 +65,43 @@ use super::CachedModule;
 ///   New version because of Metering middleware change.
 const MODULE_SERIALIZATION_VERSION: &str = "v10";
 
+/// Function that actually does the heavy lifting of creating the module version discriminator.
+///
+/// Separated for sanity tests because otherwise the `OnceLock` would cache the result.
+#[inline]
+fn raw_module_version_discriminator() -> String {
+    let wasmer_version = read_wasmer_version!();
+    let hashes = cosmwasm_vm_derive::collect_hashes();
```

Comment:
We can manually point the functions to be considered here, so we can avoid use of `inventory` dependency in this solution